Detectors/ITSMFT/ITS/simulation/src/DigitChip.cxx

```diff
@@ -0,0 +1,74 @@
+//
+//  DigitChip.cxx
+//  ALICEO2
+//
+#include "ITSSimulation/DigitChip.h"
+#include "ITSBase/Digit.h"
+
+#include "FairLogger.h"  // for LOG
+
+#include "TClonesArray.h"
+
+using namespace AliceO2::ITS;
+
+Int_t DigitChip::fnRows=650;
+
+DigitChip::DigitChip()
+{
+}
+
+DigitChip::~DigitChip()
+{
+  Reset();
+}
+
+void DigitChip::Reset()
+{
+  for (auto pixel: fPixels) {
+    delete pixel.second;
+  }
+  fPixels.clear();
+}
+
+Digit *DigitChip::FindDigit(Int_t idx)
+{
+  Digit *result = nullptr;
+  auto digitentry = fPixels.find(idx);
+  if (digitentry != fPixels.end()) {
+    result = digitentry->second;
+  }
+  return result;
+}
+
+Digit *DigitChip::GetDigit(UShort_t row, UShort_t col) {
+  Int_t idx = col*fnRows + row;
+  return FindDigit(idx); 
+}
+
+Digit *DigitChip::AddDigit(
+       UShort_t chipid, UShort_t row, UShort_t col,
+       Double_t charge, Double_t timestamp
+)
+{
+  Int_t idx = col*fnRows + row;
+
+  Digit *digit=FindDigit(idx);
+  if (digit) {
+    LOG(DEBUG) << "Adding charge to pixel..." << FairLogger::endl;
+    charge += digit->GetCharge();
+    delete digit;
+  }
+  
+  digit = new Digit(chipid,row,col,charge,timestamp);
+  fPixels.insert(std::pair<Int_t, Digit *>(idx,digit));
+
```

Comment:
Hi. I am wondering what speaks in favour of std::map here? Did you consider using an ordinary 2d vector? Maps have logarithmic access times versus O(1) for vectors. Also, if possible, we should avoid allocating new memory for each digit added. This could be achieved with a vector that preallocates space and manages the memory itself (ala TClonesArray).

Is memory consumption the problem here? How many rows and columns do we have?